From c83652816aa6641dad772f754656321e7bea037c Mon Sep 17 00:00:00 2001
From: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Date: Tue, 2 Mar 2021 19:44:22 +0800
Subject: [PATCH] arm64: fix the compile error when CONFIG_NUMA is disabled
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: c83652816aa6641dad772f754656321e7bea037c

MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

hulk inclusion
category: bugfix
bugzilla: 47994
CVE: NA

-------------------------------------------------------------------------

Fix the following compile error when CONFIG_NUMA is disabled.

arch/arm64/kernel/acpi.c: In function ‘acpi_map_cpu’:
arch/arm64/kernel/acpi.c:415:3: error: implicit declaration of function ‘numa_add_cpu’; did you mean ‘be64_add_cpu’? [-Werror=implicit-function-declaration]
   numa_add_cpu(cpu);
   ^~~~~~~~~~~~
   be64_add_cpu
arch/arm64/kernel/acpi.c: In function ‘acpi_unmap_cpu’:
arch/arm64/kernel/acpi.c:428:2: error: implicit declaration of function ‘numa_clear_node’; did you mean ‘pmd_clear_huge’? [-Werror=implicit-function-declaration]
  numa_clear_node(cpu);
  ^~~~~~~~~~~~~~~
  pmd_clear_huge

Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/kernel/acpi.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/kernel/acpi.c b/arch/arm64/kernel/acpi.c
index 0e59370bb215..b51ffac3b38d 100644
--- a/arch/arm64/kernel/acpi.c
+++ b/arch/arm64/kernel/acpi.c
@@ -31,6 +31,7 @@
 #include <asm/cputype.h>
 #include <asm/cpu_ops.h>
 #include <asm/daifflags.h>
+#include <asm/numa.h>
 #include <asm/smp_plat.h>
 
 int acpi_noirq = 1;		/* skip ACPI IRQ initialization */
-- 
2.26.2

